src/extension.c

```diff
@@ -92,41 +105,14 @@ extension_event_trigger(PG_FUNCTION_ARGS)
 	if (!CALLED_AS_EVENT_TRIGGER(fcinfo))
 		elog(ERROR, "not fired by event trigger manager");
 
+	/* TODO: does this do anything useful? */
```

Comment:
In the process of changing this code, I think you also need to revisit whether the state machine that these checks implement still makes sense. 

Now we never set DROPPING state, but I don't see you removing this state either.